topotato/assertions.py

```diff
@@ -407,6 +409,55 @@ def __call__(self):
                 )
 
 
+class AssertPing(AssertPacket):
+    _rtr: Router
+    _iface: str
+    _iface: str
+    _other_rtr: Router
+
+    @classmethod
+    def from_parent(cls, parent, name, rtr, iface, other_rtr):
+        self = super().from_parent(
+            parent, name="%s:%s/scapy[%s/ping/%s]" % (name, rtr.name, iface, other_rtr)
+        )
+
+        self._rtr = rtr
+        self._iface = iface
+        self._other_rtr = other_rtr
+        
+        rtr_ip = self._rtr.iface_to(self._iface).ip4[0].ip
+        other_rtr_ip = self._other_rtr.iface_to(self._iface).ip4[0].ip
+
+        
+        def expect_pkt(ip: IP, icmp: ICMP):
+            return ip.src == rtr_ip and ip.dst == other_rtr_ip
+                
+        self._expect_pkt = expect_pkt
+
+    def __call__(self):
+        if scapy_exc:
+            pytest.skip(scapy_exc)
+
+        router = self.instance.routers[self._rtr.name]
+        
+        rtr_ip = self._rtr.iface_to(self._iface).ip4[0].ip
+        other_rtr_ip = self._other_rtr.iface_to(self._iface).ip4[0].ip
```

Comment:
Can't do it like this for two reasons:

- the two routers won't always be connected to the same network
- the routers may have more than one IP address; for a ping test we need to give the IP not the router